modules/new_profile/jsx/NewProfileIndex.js

```diff
@@ -0,0 +1,244 @@
+        isCreated: false,
+      });
+    } else {
+      let formData = this.state.formData;
+      let formObject = new FormData();
+      for (let key in formData) {
+        if (formData[key] !== '') {
+          formObject.append(key, formData[key]);
+        }
+      }
+      fetch(this.props.submitURL, {
+        method: 'POST',
+        cache: 'no-cache',
+        credentials: 'same-origin',
+        body: formObject,
+        })
+      .then((resp) => resp.json())
+      .then((data) => {
+        this.setState({newData: data});
+        this.setState({isCreated: true});
+       });
+    }
+  }
+
+  /**
+   * Set the form data based on state values of child elements/components
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+  setFormData(formElement, value) {
+    let formData = JSON.parse(JSON.stringify(this.state.formData));
+    formData[formElement] = value;
+
+    this.setState({formData: formData});
+  }
+
+  render() {
+    // If error occurs, return a message.
+    if (this.state.error) {
+      return <h3>An error occured while loading the page.</h3>;
+    }
+
+    // Waiting for async data to load
+    if (!this.state.isLoaded) {
+      return <Loader/>;
+    }
+    let profile = null;
+    let edc = null;
+    let edcConfirm = null;
+    let project = null;
+    let pscid = null;
+    if (this.state.configData['useProject'] === 'true') {
+      project =
+        <SelectElement
+          name = "project"
+          label = "Project"
+          options = {this.state.configData.project}
+          onUserInput = {this.setFormData}
+          value = {this.state.formData.project}
+          required = {true}
+        />;
+    }
+    if (this.state.configData['edc'] === 'true') {
+      edc =
+        <DateElement
+          name = "edcDateTaken"
+          label = "Expected Date of Confinement"
+          minYear = "2000"
+          maxYear = "2017"
+          onUserInput = {this.setFormData}
+          value = {this.state.formData.edcDateTaken}
+          required = {true}
+        />;
+      edcConfirm =
+        <DateElement
+          name = "edcDateTakenConfirm"
+          label = "Confirm EDC"
+          minYear = "2000"
+          maxYear = "2017"
+          onUserInput = {this.setFormData}
+          value = {this.state.formData.edcDateTakenConfirm}
+          required = {true}
+        />;
+    }
+    if (this.state.configData['pscidSet'] === 'true') {
+      pscid =
+        <TextboxElement
+          name = "pscid"
+          label = "PSCID"
+          onUserInput = {this.setFormData}
+          value = {this.state.formData.pscid}
+          required = {true}
+        />;
+    }
+    if (!this.state.isCreated) {
+      profile =
+        <FormElement
+          name = "newProfileForm"
+          onSubmit = {this.handleSubmit} >
+        <label className = "error" >
```

Comment:
everything nested within <FormElement></FormElement> can also be indented by 2 spaces